lm_eval/api/task.py

```diff
@@ -373,13 +373,21 @@ def build_all_requests(
         world_size=None,
         cache_requests=False,
         rewrite_requests_cache=False,
+        system_instruction=None,
+        apply_chat_template=False,
+        fewshot_as_multiturn=False,
+        lm=None,
     ) -> None:
         """Build a set of Instances for a task, and store them in task.instances"""
 
         # used with caching
         og_limit = limit
 
         cache_key = f"requests-{self._config.task}-{self.config.num_fewshot}shot-rank{rank}-world_size{world_size}"
+        cache_key += "-chat_template" if apply_chat_template else ""
+        cache_key += "-fewshot_as_multiturn" if fewshot_as_multiturn else ""
+        if lm is not None and hasattr(lm, "tokenizer"):
+            cache_key += f"-{lm.tokenizer.name_or_path.replace('/', '__')}"
```

Comment:
not all tokenizers will have this attribute, unfortunately. Hm... we could also implement a second LM method that any model hoping to use chat templates will require, that returns the tokenizer or chat template's name, but not a huge fan of that...

Reply:
Sounds good to me!

Reply:
Let me try to think if I can come up with something better! if not, let's go with that

Reply:
Implemented in the suggested way, so you can check if you like it, happy to change if you see a better way!